package/MDAnalysis/analysis/pca.py

```diff
@@ -414,6 +414,174 @@ def transform(self, atomgroup, n_components=None, start=None, stop=None,
+        in the second anchor's residue, `index_extrapolate` is [0, 0, 1, 1, 1]
+
+        Examples
+        --------
+        Run PCA class before using this function. For backbone PCA, run::
+
+            pca = PCA(universe, select='backbone').run()
+
+        Obtain a transformation function to project the
+        backbone trajectory onto the first principal component::
+
+            project = pca.project_single_frame(components=0)
+
+        To project onto the first two components, run::
+
+            project = pca.project_single_frame(components=[0,1])
+
+        Alternatively, the transformation can be applied to PCA atoms and
+        extrapolated to other atoms according to the CA atom's translation
+        in each residue::
+
+            all = u.select_atoms('all')
+            project = pca.project_single_frame(components=0,
+                                               group=all, anchor='name CA')
+
+        Finally, apply the transformation function to a timestep::
+
+            project(u.trajectory.ts)
+
+        or apply the projection to the universe::
+
+            u.trajectory.add_transformations(project)
+
+
+        .. versionadded:: 2.2.0
+        """
+        if not self._calculated:
+            raise ValueError('Call run() on the PCA before projecting')
+
+        if group is not None:
+            if anchor is None:
+                raise ValueError("'anchor' cannot be 'None'" +
+                                 " if 'group' is not 'None'")
+
+            anchors = group.select_atoms(anchor)
+            anchors_res_ids = anchors.resindices
+            if np.unique(anchors_res_ids).size != anchors_res_ids.size:
+                raise ValueError("More than one 'anchor' found in residues")
+            if not np.isin(group.resindices, anchors_res_ids).all():
+                raise ValueError("Some residues in 'group'" +
+                                 " do not have an 'anchor'")
+            if not anchors.issubset(self._atoms):
+                raise ValueError("Some 'anchors' are not part of PCA class")
+
+            # non_pca has "all" the atoms in residues of `group`. This makes
+            # sure that extrapolation works on residues, not random atoms.
+            non_pca = group.residues.atoms - self._atoms
+            pca_res_indices, pca_res_counts = np.unique(
+                self._atoms.resindices, return_counts=True)
+
+            non_pca_atoms = np.array([], dtype=np.int64)
+            for res in group.residues:
+                # n_common is the number of pca atoms in a residue
+                n_common = pca_res_counts[np.where(
+                           pca_res_indices == res.resindex)][0]
+                non_pca_atoms = np.append(non_pca_atoms,
+                                          res.atoms.n_atoms - n_common)
+            # index_extrapolate records the anchor number for each non-PCA atom
+            index_extrapolate = np.repeat(np.arange(anchors.atoms.n_atoms),
```

Comment:
There's one failure in the 32 bit windows run https://dev.azure.com/mdanalysis/mdanalysis/_build/results?buildId=3458&view=logs&j=8c1244f7-4387-51eb-cf10-2a1290f32f77&t=6cc8a0ec-9d5a-5758-2f06-db7cff0afb7f

There's an implicit int64 -> int32 case (maybe in repeat()) that needs to be made explicit.

Reply:
Probably fixed now.

If I understand correctly, it was because I had an array explicitly defined as np.int64, which later required an implicit conversion on a 32-bit machine. I changed the array creation line to int. The new declared array should not require a conversion later.

If the tests are being run in the same order, the new code has passed the test. I'll edit the comment when the test is done.

Reply:
you solved it – test passes!